client/src/pages/OfferForm/index.tsx

```diff
@@ -0,0 +1,106 @@
+import { useState } from 'react'
+import './OfferForm.css'
+import { OfferData } from '../../types/offerTypes'
+
+interface OfferFormProps {
+  task: {
+    id: string
+    title: string
+  }
+  onSubmit: (offerData: OfferData) => void
+  onCancel: () => void
+}
+
+const OfferForm = ({ task, onSubmit, onCancel }: OfferFormProps) => {
+  const [offerData, setOfferData] = useState<OfferData>({
+    offer_description: '',
+    offer_media: null,
+    posting_time: '',
+    price: '',
+    offer_status: 'pending',
+  })
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    setOfferData({
+      ...offerData,
+      [event.target.name]: event.target.value,
+    })
+  }
+
+  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0] || null
+    setOfferData({
+      ...offerData,
+      offer_media: file,
```

Comment:
nit: You can consider allowing an array of files to represent multiple media.
```javascript
  const handleFileChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const files = event.target.files;
    if (files) {
      setOfferData((prevData) => ({
        ...prevData,
        offer_media: Array.from(files),
      }));
    }
  };
```

Reply:
Nice one!